### DISCUSSION TITLE
Fastify with InversifyJS

### DISCUSSION THREAD

**Discussion by leonardssh** [fastify/fastify#2879] [Q&A] [answered]
Is there a library or something to integrate [**InversifyJS**](https://inversify.io/) with **Fastify**? (like [inversify-express-utils](https://github.com/inversify/inversify-express-utils))

If not, I could create one, based on ideas from [inversify-express-utils](https://github.com/inversify/inversify-express-utils). 😊

![hmm](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fmedia.giphy.com%2Fmedia%2F7K3p2z8Hh9QOI%2Fgiphy.gif&f=1&nofb=1)

**Comment by jsumners** [comment, marked as answer]
If you haven't found one on `npmjs.com` or in our ecosystem doc, then I think it is safe to say no one has created it yet.

**Comment by kibertoad** [comment]
There is one for awilix if you need DI: https://github.com/fastify/fastify-awilix
Any reasons why you need Inversify specifically?

**Reply by leonardssh** [reply]
Yeah.. not really, but I prefer InversifyJS instead of anything else. (personal preferences). 😊

But you made me curious about Awilix, I hadn't heard of him before, I think I'll give him a chance as soon. 

Thanks!
